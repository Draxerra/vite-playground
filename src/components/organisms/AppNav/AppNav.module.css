.app-nav {
  display: flex;
  flex-wrap: wrap;
}
