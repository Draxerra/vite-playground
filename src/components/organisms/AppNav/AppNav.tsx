import React, { FC } from "react";
import classNames from "classnames";

import AppLinkButton from "~/components/molecules/AppLinkButton";

import styles from "./AppNav.module.css";

type AppNavProps = JSX.IntrinsicElements["nav"];
const AppNav: FC<AppNavProps> = ({ className, ...props }) => (
  <nav {...props} className={classNames(className, styles.appNav)}>
    <h2 className="sr-only">Site Navigation</h2>
    <AppLinkButton to="/" aria-label="Navigate to Home Page">
      Home
    </AppLinkButton>
    <AppLinkButton to="/contact" aria-label="Navigate to Contact Page">
      Contact
    </AppLinkButton>
  </nav>
);

export default AppNav;
