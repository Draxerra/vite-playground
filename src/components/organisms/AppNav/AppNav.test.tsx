import React from "react";
import AppNav from "./AppNav";
import { BrowserRouter as Router } from "react-router-dom";
import { render } from "@testing-library/react";

describe("App Nav", () => {
  it("should match snapshot", () => {
    const nav = render(<AppNav />, { wrapper: Router }).asFragment();
    expect(nav).toMatchSnapshot();
  });
});
