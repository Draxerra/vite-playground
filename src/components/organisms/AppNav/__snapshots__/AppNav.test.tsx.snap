// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Nav should match snapshot 1`] = `
<DocumentFragment>
  <nav
    class="appNav"
  >
    <h2
      class="sr-only"
    >
      Site Navigation
    </h2>
    <a
      aria-current="page"
      aria-label="Navigate to Home Page"
      class="appLinkButton appLinkButtonActive appButton"
      href="/"
    >
      Home
    </a>
    <a
      aria-label="Navigate to Contact Page"
      class="appLinkButton appButton"
      href="/contact"
    >
      Contact
    </a>
  </nav>
</DocumentFragment>
`;
